src/i_video.c

```diff
@@ -428,18 +427,21 @@ static void I_ToggleFullScreen(void)
     {
         SDL_SetWindowMouseGrab(screen, true);
         SDL_SetWindowResizable(screen, false);
+        SDL_SetWindowFullscreen(screen, true);
+        SDL_SyncWindow(screen);
         SDL_SetWindowFullscreenMode(screen, NULL);
     }
     else
     {
+        SDL_SetWindowFullscreen(screen, false);
+        SDL_SyncWindow(screen);
         SDL_SetWindowMouseGrab(screen, false);
         SDL_SetWindowResizable(screen, true);
         SDL_SetWindowBordered(screen, true);
         AdjustWindowSize();
         SDL_SetWindowSize(screen, window_width, window_height);
     }
 
-    SDL_SetWindowFullscreen(screen, fullscreen);
     SDL_SyncWindow(screen);
```

Comment:
Why didn't you remove the `SDL_SyncWindow` call? Perhaps we don't need this function at all. It seems to do nothing on Windows. It might work on Linux with Wayland, though.